### PR TITLE
Feature/appc credits contract

### PR DESCRIPTION
**Pull request by GustavoRSSilva**
## Checklist:

- [x] You have followed our [**contributing guidelines**](https://github.com/AppStoreFoundation/asf-contracts/wiki/Contributing-to-AppCoins-Protocol-smart-contracts)
- [x] double-check your branch is based on `dev` and targets `dev` 
- [x] Pull request has tests (at least 80% coverage)
- [x] Code is well-commented, linted and follows project conventions
- [x] Documentation is updated (if necessary)
- [x] Description explains the issue/use-case resolved and auto-closes related issues

### CODE REVIEW

File: contracts/AppCoinsCreditsBalance.sol [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        return balance;
+    }
+
+    /**
+    *@notice Get the balance proof
+    *@dev
+    *    returns the balance proof
+    *@return {"balanceProof" : "balance proof"}
+    */
+    function getBbalanceProof() public view returns(bytes) {
```
Comment (tcrosado) [inline contracts/AppCoinsCreditsBalance.sol]: I think this is a typo

File: contracts/AppCoinsCreditsBalance.sol [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        return balanceProof;
+    }
+
+
+
+    function registerBalanceProof(bytes _merkleTreeHash)
```
Comment (tcrosado) [inline contracts/AppCoinsCreditsBalance.sol:53]: This function should have some documentation

File: contracts/AppCoinsCreditsBalance.sol [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        public
+        onlyIfWhitelisted("registerBalanceProof",msg.sender){
+
+        balanceProof = _merkleTreeHash;
+
+        emit BalanceProof(_merkleTreeHash);
+    }
+
+    function depositFunds(uint _amount)
```
Comment (tcrosado) [inline contracts/AppCoinsCreditsBalance.sol]: Talking with @zepcp, registerBalanceProof can be an internal function called when depositFunds and withdrawFunds functions are called

**Review by tcrosado — CHANGES_REQUESTED** [review]

**Review by tcrosado — APPROVED** [review]

File: contracts/Advertisement.sol [hunk 4]
```diff
@@ -41,8 +41,8 @@ contract Advertisement is BaseAdvertisement {
             uint[] vercodes
```
Comment (DioPires) [inline contracts/Advertisement.sol:41]: #### Tests:
```diff 
+ 112 passing
```
#### Test coverage: 
82.97 %

Comment (DioPires) [inline contracts/Advertisement.sol:41]: #### Tests:
```diff 
+ 112 passing
```
#### Test coverage: 
82.97 %

Comment (DioPires) [inline contracts/Advertisement.sol:41]: #### Tests:
```diff 
+ 112 passing
```
#### Test coverage: 
82.97 %

Comment (DioPires) [inline contracts/Advertisement.sol:41]: #### Tests:
```diff 
+ 112 passing
```
#### Test coverage: 
82.97 %

Comment (DioPires) [inline contracts/Advertisement.sol:41]: #### Tests:
```diff 
+ 112 passing
```
#### Test coverage: 
82.97 %

**Review by DioPires — APPROVED** [review]

**Pull request merged**